learning/tour-of-beam/frontend/lib/pages/tour/state.dart

```diff
@@ -218,7 +295,12 @@ class TourNotifier extends ChangeNotifier with PageStateMixin<void> {
     _unitContentCache.removeListener(_onUnitChanged);
     contentTreeController.removeListener(_onUnitChanged);
     _appNotifier.removeListener(_onAppNotifierChanged);
-    _authNotifier.removeListener(_onUnitProgressChanged);
+    _authNotifier.removeListener(_onAuthChanged);
+    playgroundController.snippetEditingController
+        ?.removeListener(_snippedEditingControllerListener);
+    playgroundController
+        .snippetEditingController?.activeFileController?.codeController
+        .removeListener(_saveCodeListener);
```

Comment:
Look like we will still have listeners on all `CodeController`s we ever viewed except the one that is active when closing this screen.
This emphasizes the need for events and listening to `PlaygroundController` only. Please file an issue and add it in TODO here.

Reply:
https://github.com/apache/beam/issues/25185